### PR TITLE
Set private browsing permission to enable FF os menuitem in private browsing only sessions fixes #431

### LINKED ISSUE #431: can't find Addon when 'History' is set to ' Never remember history' in preferences
seperate from #385 because it's absolutely different bug
### details

I have problem find Addon at Nightly 22.0a1(not nightly anymore) + FxOS Simulator 3.0pre4.
I'm now using 23.0a1 + FxOs Simulator 3.0pre6 and still confirm this issue, but seems only me.
I use Firefox with 'Never remember history' and when I change preference and restart, I can see Addon back.

It's hard to find people who set preference like me, so it makes hard to find what makes this bug.
Also I guess this bug will happen in Aurora 22, but didn't test it.

### PR DESCRIPTION
**Pull request by ochameau**

### CODE REVIEW

**Comment by mykmelez** [comment]
@ochameau does this also fix #246? I.e. does it work around [SDK bug 855484](https://bugzilla.mozilla.org/show_bug.cgi?id=855484)?

**Comment by ochameau** [comment]
Yes and no. It ends up doing something weird.
The menu now does appear, but in the last non-private window instead of the newly opened private window.

**Comment by ochameau** [comment]
Let me rephrase it, the meny appears but when you click on it, the simulator is opened in a tab in the non-private window in the background.
